api/converter_utils.go

```diff
@@ -283,12 +287,7 @@ func txnRowToTransaction(row idb.TxnRow) (generated.Transaction, error) {
 		Sig:      sigToTransactionSig(stxn.Sig),
 	}
 
-	var txid string
-	if row.Extra.RootIntra.Present {
-		txid = row.Extra.RootTxid
-	} else {
-		txid = crypto.TransactionIDString(stxn.Txn)
-	}
+	txid := crypto.TransactionIDString(stxn.Txn)
```

Comment:
Unfortunately, this will change the behavior on how inner txn txids are returned in this API: https://github.com/algorand/indexer/blob/develop/api/handlers.go#L752

It feels like the intended behavior is for the inner txid to always match the root txid, especially since inner txns have different logic to calculate their "ID"s. I added a test here to check this TxID behavior: https://github.com/winder/indexer/pull/2

Reply:
AFAIK inner transactions aren't returned in the API. Are they returned in the logs endpoint?

Reply:
The log endpoint response returns the logs and the TxID that logged it: https://github.com/algorand/indexer/blob/develop/api/generated/common/types.go#L249

So if an inner txn calls app #12345 that logs something, then the log response for app #12345 would have an array of `{log(s)-TxID}`, where TxID is currently the root TxID.

Reply:
Thanks for the explanation, and the test case. I'll revert this part.